pkg/reconciler/revision/background.go

```diff
@@ -64,8 +64,12 @@ type resolveResult struct {
 	// rather than an array so that we can easily determine when we have received
 	// all the digests that we want (by comparing length to `want`).
 	statuses map[int]v1.ContainerStatus
-	want     int
-	err      error
+
+	// initContainerStatuses like `statuses` is a map of init container index to resolved status.
+	initContainerStatuses map[int]v1.ContainerStatus
```

Comment:
an alternative approach might be to keep just one internal containerStatuses field, and build the two result arrays at the end of Resolve. That might simplify the logic a bit (we could drop isInitContainer and related if/else branches, I think). I'd be OK to land this as is and I can try that refactor in a follow-up, though.

Reply:
I think it was just very slightly easier originally to do it with an int since that way you can build up the array and return it / I didn't think of using a string lookup when I wrote v1 of this :). I'll happily dig back in to this and refactor though once this lands (unless @skonto particularly wants to)

Reply:
I didnt want to create an array where I have a convention about ordering here. We agreed not to have it at the api level as it is simpler when statuses are reported in separate manner and my thought was that it is also better to keep it simple internally for readability and easy mapping of the values to the two status fields. I am not sure if there is a big gain here, but I will refactor no need to do it in a separate one.